test/e2e/federate.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	ctlutil "github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	"github.com/kubernetes-sigs/federation-v2/pkg/kubefed2/federate"
+	"github.com/kubernetes-sigs/federation-v2/test/common"
+	"github.com/kubernetes-sigs/federation-v2/test/e2e/framework"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	"sort"
+)
+
+var _ = Describe("Federate", func() {
+	f := framework.NewFederationFramework("federate-resource")
+	tl := framework.NewE2ELogger()
+	typeConfigFixtures := common.TypeConfigFixturesOrDie(tl)
+
+	var kubeConfig *restclient.Config
+	var testNamespace, systemNamespace string
+	var clusterNames []string
+
+	BeforeEach(func() {
+		if kubeConfig == nil {
+			kubeConfig = f.KubeConfig()
+			testNamespace = f.TestNamespaceName()
+			systemNamespace = f.FederationSystemNamespace()
+			clusterNames = f.ClusterNames("federate-resource-test")
+		}
+	})
+
+	for key := range typeConfigFixtures {
+		typeConfigName := key
+		fixture := typeConfigFixtures[key]
+		Describe(fmt.Sprintf("resource for federated type %q", typeConfigName), func() {
+			It("should create a equivalant federated resource in federation", func() {
+				if typeConfigName == util.NamespaceName {
+					// TODO(irfanurrehman): Update tests after implementing federate namespaces
+					framework.Skipf("Federate %s is not supported yet.", typeConfigName)
+				}
+
+				dynClient, err := client.New(f.KubeConfig(), client.Options{})
+				if err != nil {
+					tl.Fatalf("Error initializing dynamic client: %v", err)
+				}
+				typeConfig, err := common.GetTypeConfig(dynClient, typeConfigName, systemNamespace)
+				if err != nil {
+					tl.Fatalf("Error retrieving typeconfig %q: %v", typeConfigName, err)
+				}
+
+				// The test namespace should exist in target cluster for namespaced resources
+				if typeConfig.GetNamespaced() {
```

Comment:
Why is this necessary?  Federate is specific to the host cluster.

Reply:
This was not supposed to be there; missed removing it in the first pass. Removed now!